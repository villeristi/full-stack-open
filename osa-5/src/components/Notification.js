import React from 'react'
import './notification.css'

export default ({ msg, status }) => {
  const className = status && status !== 'success' ? 'error' : 'success'

  if(!msg) {
    return null
  }

  return (
    <div className={`notification ${className}`}>
      <p>{msg}</p>
    </div>
  )
}
