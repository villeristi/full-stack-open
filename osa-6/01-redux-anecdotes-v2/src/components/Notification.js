import React from 'react'
import { connect } from 'react-redux'

class Notification extends React.Component {
  render() {
    const { notification: { message } } = this.props

    const style = {
      border: 'solid',
      padding: 10,
      borderWidth: 1,
      marginBottom: 10,
    }

    if(!message)Â {
      return null
    }

    return (
      <div style={style}>
        {message}
      </div>
    )
  }
}

export default connect((state) => {
  return {
    notification: state.notification
  }
})(Notification)

