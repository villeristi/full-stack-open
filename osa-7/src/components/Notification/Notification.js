import React from 'react'
import './notification.css'

export default ({ notification }) => {

  const { message, status } = notification

  if(!message) {
    return null
  }

  const className = status && status !== 'success' ? 'warning' : 'success'

  return (
    <div className={`mt-3 alert alert-${className}`} role="alert">
      {message}
    </div>
  )
}
