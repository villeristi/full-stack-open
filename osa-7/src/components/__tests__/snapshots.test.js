import React from 'react';
import Notification from '../Notification/Notification';
import renderer from 'react-test-renderer';

describe('Snapshots', () =>Â {
  describe('<Notification />', () => {
    it('renders correctly', () => {

      const notif = {
        message: 'Test',
        status: 'error'
      }

      const tree = renderer
        .create(<Notification notification={notif} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    })
  })
})
