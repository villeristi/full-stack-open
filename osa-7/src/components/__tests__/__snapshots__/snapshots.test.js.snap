// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots <Notification /> renders correctly 1`] = `
<div
  className="mt-3 alert alert-warning"
  role="alert"
>
  Test
</div>
`;
